tests/robustness/coverage/README.md

```diff
@@ -0,0 +1,123 @@
+## Overview
+
+This is still Work In Progress.
+
+Go tests in this directory analyze the usage of Etcd API based on collected
+traces from a running cluster. They will output information on:
+
+1. Traces that **did go** through the contract interface:
+   1. Get (split by: `Revision` set)
+   1. OptimisticPut (split by: `ExpectedRevision` set, `Lease` set, `GetOnFailure` set)
+   1. OptimisticDelete (split by: `ExpectedRevision` set, `GetOnFailure` set)
+   1. List (split by: `Revision` set, `Limit` set or one or zero)
+1. Traces that **did not go** through the contract interface:
+   1. Known uses of the interface within `KV`:
+      1. Healthchecks (reads to `/registry/health`)
+      1. Counts
+      1. Compact checks (i.e. reads that use `compact_rev_key`)
+      1. Consistent Reads
+   1. Known uses of the interface outside `KV`:
+      1. Calls to `Lease/LeaseGrant` (used for master leases)
+      1. Calls to `Maintenance/Status` (used to get information on database size)
+      1. Calls to `Watch/Watch` (as it is not part of the contract interface yet)
+
+### Manual test execution
+
+At first we will manually set up the cluster, run e2e tests, download traces and
+then execute the test.
+
+0\. The commands from 1\. to 3\. below are expected to be run in a `kubernetes`
+repository patched with
+https://github.com/AwesomePatrol/kubernetes/commit/5a18a66c06bad4350fb1fd52b3ca50ab885dbbf1
+or
+https://github.com/AwesomePatrol/kubernetes/tree/add-kubernetes-etcd-contract-tracker
```

Comment:
This PR is mostly about determining if the request falls into K8s interface. Could we focus on collecting and classifying the requests made by K8s to etcd first, and maybe do interface classification as next step?